pkg/dataobj/consumer/service.go

```diff
@@ -150,54 +153,60 @@ func (s *Service) handlePartitionsRevoked(partitions map[string][]int32) {
 }
 
 func (s *Service) run(ctx context.Context) error {
-	for {
-		fetches := s.client.PollRecords(ctx, -1)
-		if fetches.IsClientClosed() || ctx.Err() != nil {
-			return nil
-		}
-		if errs := fetches.Errors(); len(errs) > 0 {
-			var multiErr error
-			for _, err := range errs {
-				multiErr = errors.Join(multiErr, err.Err)
-			}
-			level.Error(s.logger).Log("msg", "error fetching records", "err", multiErr.Error())
-			continue
-		}
-		if fetches.Empty() {
-			continue
-		}
+	g, ctx := errgroup.WithContext(ctx)
```

Comment:
I don't understand why this needs to run in a goroutine, there is only a single one, and there is no other computation in parallel.